Src/FluentAssertions/Specialized/FunctionAssertions.cs

```diff
@@ -11,53 +11,25 @@ namespace FluentAssertions.Specialized
     /// Contains a number of methods to assert that a synchronous function yields the expected result.
     /// </summary>
     [DebuggerNonUserCode]
-    public class FunctionAssertions<T> : ReferenceTypeAssertions<Func<T>, FunctionAssertions<T>>
+    public class FunctionAssertions<T> : DelegateAssertions<Func<T>>
     {
-        private readonly IExtractExceptions extractor;
-
-        public FunctionAssertions(Func<T> subject, IExtractExceptions extractor)
+        public FunctionAssertions(Func<T> subject, IExtractExceptions extractor) : base(subject, extractor)
         {
-            this.extractor = extractor;
             Subject = subject;
```

Comment:
🙃 Redundant assignment?

Reply:
Not redundant, as this initialises the new property which hides the base definition on line 24. 
This means we have a value which is used for populating both a Func<T> and Delegate property.